sky/cli.py

```diff
@@ -3687,11 +3687,17 @@ def jobs_launch(
     dag_utils.maybe_infer_and_fill_dag_and_task_names(dag)
     dag_utils.fill_default_config_in_dag_for_job_launch(dag)
 
-    click.secho(f'Managed job {dag.name!r} will be launched on (estimated):',
-                fg='cyan')
     dag, _ = admin_policy_utils.apply(
         dag, use_mutated_config_in_current_request=False)
-    dag = sky.optimize(dag)
+
+    if not yes:
+        # Skip resource preview if -y is set, since we are probably running in
+        # a script and the user won't have a chance to review it anyway.
+        # This can save a couple of seconds.
+        click.secho(
+            f'Managed job {dag.name!r} will be launched on (estimated):',
+            fg='cyan')
+        dag = sky.optimize(dag)
```

Comment:
Updated:
```
cooperc@beacon ~/a/skypilot (launch-optimizations) [1]> sky jobs launch -y echo test
Task from command: echo test
Resources for managed job 'sky-cmd' will be computed on the managed jobs controller, since --yes is set.
Launching managed job 'sky-cmd' from jobs controller...
Choosing resources for managed jobs controller...
Considered resources (1 node):
----------------------------------------------------------------------------------------------
 CLOUD   INSTANCE          vCPUs   Mem(GB)   ACCELERATORS   REGION/ZONE   COST ($)   CHOSEN
----------------------------------------------------------------------------------------------
 AWS     m6i.2xlarge       8       32        -              us-east-1     0.38          ✔
 Azure   Standard_D8s_v5   8       32        -              eastus        0.38          
----------------------------------------------------------------------------------------------

```